crates/sui-core/src/authority.rs

```diff
@@ -1273,6 +1305,53 @@ impl AuthorityState {
             .inc();
     }
 
+    /// Waits for fastpath (owned) input objects to become available until the specified max_wait timeout.
+    /// Returns true if all fastpath input objects are available, false otherwise.
+    pub(crate) async fn wait_for_fastpath_input_objects(
+        &self,
+        transaction: &VerifiedTransaction,
+        epoch: EpochId,
+        max_wait: Duration,
```

Comment:
nit: `max_wait` -> `timeout_duration` ? I see that in the rest of the code we pass variables with name like `wait_for_fastpath_input_objects_timeout` , which kind of dictates that this is a timeout